libs/@blockprotocol/service/README.md

```diff
@@ -0,0 +1,63 @@
+## Block Protocol – Service Module
+
+This package implements the Block Protocol Service module for blocks and embedding applications.
+
+To get started:
+
+1.  `yarn add @blockprotocol/service` or `npm install @blockprotocol/service`
+1.  Follow the instructions to use the service module as a [block](#blocks) or an [embedding application](#embedding-applications)
+
+## Blocks
+
+To create a `ServiceBlockHandler`, pass the constructor an element in your block, along with any callbacks you wish to register to handle incoming messages.
+
+### React example
+
+For React, we provide a `useServiceBlockService` hook, which accepts a `ref` to an element, and optionally any `callbacks` you wish to provide on initialization.
+
+See `npx create-block-app@latest my-block --template react` for an example.
+
+### Custom elements
+
+<!-- TODO - https://app.asana.com/0/0/1204003986334469/f -->
```

Comment:
I don't think we need a separate base class for the service module on top of the graph one – don't think a class per module is scalable in the same way as a hook per module is.
```suggestion
Create a `ServiceBlockHandler` in your element. If you are using our template, you can add this in the constructedCallback instance method ([Lit docs](https://lit.dev/docs/components/lifecycle/#connectedcallback)) and make it available for use in your element.

```ts
  ...

  connectedCallback() {
    super.connectedCallback();
    if (!this.serviceModule || this.serviceModule.destroyed) {
      this.serviceModule = new ServiceBlockHandler({
        element: this,
      });
    }
  }
  
  retrieveDirections(data) {
    if (!this.serviceModule) {
      throw new Error("Service module not available");
    }
    this.serviceModule.mapboxRetrieveDirections(args).then(() => { ... });
  } 
  
  ...
```
```

Reply:
Ah that makes sense, thank you Ciaran! Have archived [the follow-up task](https://app.asana.com/0/0/1204003986334469/f) and applied your suggestion in https://github.com/blockprotocol/blockprotocol/pull/954/commits/76f2712f20c23737a2820fda00631ccec393e291